docs/readmes/README.best-practices.md

```diff
@@ -0,0 +1,206 @@
+# Best Practices
+
+This document highlights the best practices when using aad-pod-identity.
+
+## Retry on token retrieval
+
+aad-pod-identity retrieves AAD tokens on behalf of your workload by intercepting token requests to the Instance Metadata Service (IMDS) endpoint (169.254.169.254). However, it does not perform any sort of token caching and retries. We believe that either the users or the SDK that your application is using is responsible for token management.
+
+Note that there is a brief window between when your application started running and the identity gets assigned to the underlying node. If your application tried to retrieve a token during that period, it will fail. The following table describes the duration that you could expect based on your cluster's node type:
```

Comment:
We should also add a note about the in built NMI retries we have in place. If the token request times out before these retries succeed, then token it'll fail. Also, add a note that this is only the first time the identity is being assigned to the underlying node. Subsequent token requests will not incur this delay.